### DISCUSSION TITLE
RectResize.model 可以通过除了监听缩放事件以外的途径通过缩放触发一个函数吗

### DISCUSSION THREAD

**Discussion by cpf23333** [didi/LogicFlow#1288] [Q&A] [answered]
如题，监听事件需要获取到LogicFlow实例并将事件中newNodeSize的id与类实例的id做比较，然后才能确认事件是本节点触发的
有没有另外的途径由本节点的缩放直接触发本节点类里定义的一个方法

**Comment by wumail** [comment]
试试继承并重写resize方法

**Reply by cpf23333** [reply]
![image](https://github.com/didi/LogicFlow/assets/39972632/921a1d73-b182-42a4-9f9b-62a4a3e19d03)

这个方法没有在文档中提到，实现也只是一个log，我重写了这个方法，没有被触发

是用来被外部调用的吗

我这里的需求是由用户来拖拽节点四角的锚点进行缩放

**Reply by wumail** [reply, marked as answer]
需要手动调用一下，大概是这个意思，类里面的任何方法都可以，不一定是resize
```typescript
lf.on('node:resize', ({ oldNodeSize, newNodeSize }: any) => {
 const { id } = newNodeSize
 lf.getNodeModelById(id).resize(newNodeSize)
})

class model extends GroupNode.model{
  ...
  resize(newNodeSize){
    const { x , y, width, height } = newNodeSize
    this.x = x + width / 2 
    this.y = y + height / 2
    // do anything you want
  }
  ...
}
```

**Reply by cpf23333** [reply]
lf上监听，这个就是我现在的实现方式😂
看样子现在没有第二种办法了，就按这种方式实现吧
